File: Remove_Duplicates_from_Sorted_Array.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        
        index = 1
        n = len(nums)
        for cur in range(n-1):
            if nums[cur] != nums[cur+1]:
                nums[index] = nums[cur+1]
                index += 1
        return index
            
        
        
        
        
        
        
        # cur = 0
        # fdup = 0
        # while fdup < len(nums) :  
        #     while (fdup+1) < len(nums) and nums[fdup] == nums[fdup+1] :
        #         fdup += 1
        #     if fdup == len(nums)-1:
        #         break
        #     fdup += 1
        #     cur += 1
        #     nums[cur] = nums[fdup]
        # return cur+1  

File: Remove_Duplicates_from_Sorted_Array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if (nums.size() == 0) return 0;
        if (nums.size() == 1) return 1;
        
        
        int cur = 0, move = 1;
        
        while(move < nums.size()){
            if (nums[move] != nums[cur])
                nums[++cur] = nums[move];
            move++;
            
        }
        return cur + 1;
    }
};